fabric-extensibility-api-v1/src/main/java/net/fabricmc/fabric/mixin/extensibility/BowItemMixin.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.extensibility;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.item.ArrowItem;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.BowItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.extensibility.item.v1.FabricBowHooks;
+
+@Mixin(BowItem.class)
+public class BowItemMixin {
+	@Inject(method = "onStoppedUsing(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/entity/LivingEntity;I)V", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;spawnEntity(Lnet/minecraft/entity/Entity;)Z"), locals = LocalCapture.CAPTURE_FAILHARD)
+	public void onStoppedUsing(ItemStack stack, World world, LivingEntity user, int remainingUseTicks, CallbackInfo info, PlayerEntity playerEntity, boolean bl, ItemStack itemStack, int i, float f, boolean bl2, ArrowItem arrowItem, PersistentProjectileEntity persistentProjectileEntity) {
+		if (this instanceof FabricBowHooks) {
+			((FabricBowHooks) this).onBowRelease(stack, user, remainingUseTicks, persistentProjectileEntity);
```

Comment:
This mixin seems really simple and thus kind of unnecessary in fapi. Like if the FabricBowHooks interface only exists in order to create this method, and the mixin for doing so is this simple (and its just an inject so multiple mods can do it), then I don't know if its really necessary to have this in here.

That being said, if we want a reason to have it in here I think it would be a good idea to make an event callback that can be subscribed to when a bow is released. So create an event that we can register callbacks to and allow each callback in the event to modify the arrow entity or cancel the event altogether.

Reply:
This method was more for modifying the arrow entity, not really an event